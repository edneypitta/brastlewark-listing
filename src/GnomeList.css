ul.gnome-list {
  width: 50vh;
  margin: 0 auto;
}

ul.gnome-list li {
  list-style: none;
  display: flex;
  align-items: center;
  margin-bottom: 20px;
  font-size: calc(10px + 2vmin);
  text-decoration: none;
}

ul.gnome-list li:hover {
  cursor: pointer;
  text-decoration: underline;
}

ul.gnome-list li :nth-child(1) {
  margin-right: 20px;
}