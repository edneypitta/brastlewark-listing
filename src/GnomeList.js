import React from 'react'
import PropTypes from 'prop-types'
import './GnomeList.css'
import Avatar from './shared/Avatar'

const GnomeList = ({ gnomes, onClick }) =>
  <ul className="gnome-list">
    {gnomes.map(gnome =>
      <li key={gnome.id} onClick={onClick.bind(null, gnome)}>
        <div>
          <Avatar
            src={gnome.thumbnail}
            alt="Gnome Thumbnail" />
        </div>
        <div>{gnome.name}</div>
      </li>
    )}
  </ul>

GnomeList.propTypes = {
  gnomes: PropTypes.array.isRequired,
  onClick: PropTypes.func.isRequired
}

export default GnomeList