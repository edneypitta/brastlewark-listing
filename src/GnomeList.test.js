import React from 'react'
import { shallow } from 'enzyme'
import GnomeList from './GnomeList'

describe('GnomeList', () => {
  it('renders without crashing', () => {
    const component = shallow(
      <GnomeList
        gnomes={[{ name: 'gnome 1' }, { name: 'gnome 2' }]}
        onClick={jest.fn()} />)
    expect(component).toMatchSnapshot()
  })
})
