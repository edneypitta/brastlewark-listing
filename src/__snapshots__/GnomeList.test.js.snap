// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GnomeList renders without crashing 1`] = `
<ul
  className="gnome-list"
>
  <li
    onClick={[Function]}
  >
    <div>
      <Avatar
        alt="Gnome Thumbnail"
      />
    </div>
    <div>
      gnome 1
    </div>
  </li>
  <li
    onClick={[Function]}
  >
    <div>
      <Avatar
        alt="Gnome Thumbnail"
      />
    </div>
    <div>
      gnome 2
    </div>
  </li>
</ul>
`;
